### ISSUE TITLE
Can't reinstall an app

### ISSUE THREAD

**Issue by amo13** [mvdan/fdroidcl#24]
Hey, I am probably just doing something wrong, but could you please help me with this issue? I am on archlinux with a Sony Xperia SP connected. I successfully managed to install the app waveup, then  uninstall it (did not start it in between), but now I can't install it again as I get this strange errors...

```
[amo@AmoNotebook ~]$ fdroidcl update
Downloading https://f-droid.org/repo/index.jar... not modified
[amo@AmoNotebook ~]$ fdroidcl install com.jarsilio.android.waveup
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x565c21]

goroutine 1 [running]:
github.com/mvdan/fdroidcl.(*Apk).URL(0xc4205e1c00, 0x0, 0x3)
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/index.go:203 +0x51
main.downloadApk(0xc4205e1c00, 0xc4200d6700, 0xc4205e1c00)
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/cmd/fdroidcl/download.go:40 +0x40
main.downloadAndDo(0xc420096018, 0x1, 0x1, 0xc4200d6700, 0x734f40)
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/cmd/fdroidcl/install.go:49 +0xc4
main.runInstall(0xc42007c1d0, 0x1, 0x1)
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/cmd/fdroidcl/install.go:35 +0x1cc
main.main()
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/cmd/fdroidcl/main.go:180 +0x384
```

**Comment by mvdan** [comment]
This does seem like a bug. What version of `fdroidcl` are you running? Did you try the latest master?

**Comment by mvdan** [comment]
This is likely https://github.com/mvdan/fdroidcl/issues/7. Did you try wiping your index cache, i.e. `~/.cache/fdroidcl`?

This is because older versions of `fdroidcl` wrote a bad cache on disk. Not much I can do to fix that now, I'm afraid.

**Comment by amo13** [comment]
Thanks for looking into this! According to the Arch User Repository (where I installed it from) I am on version 0.3.0-1.
I just tried deleting all files in ~/.cache/fdroidcl, but the problem persists. Please let me know if I can do anything to help solving this!
And by the way: thank you A LOT for this great tool! I flash new roms on many different devices recently and this is such a great and time-saving help to me!

**Comment by mvdan** [comment]
@amo13 could you try the latest master?

    go get -u github.com/mvdan/fdroidcl/cmd/fdroidcl

**Comment by mvdan** [comment]
In particular, there also was #19 and the fix is not in 0.3.0. I meant to do a 0.3.1 but never got to it.

**Comment by amo13** [comment]
ok, I do not really understand how go works, how to compile and stuff like that, but I executed your command. unfortunately it did not help:
```
[amo@AmoNotebook ~]$ go get -u github.com/mvdan/fdroidcl/cmd/fdroidcl
[amo@AmoNotebook ~]$ fdroidcl install com.jarsilio.android.waveuppanic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x565c21]

goroutine 1 [running]:
github.com/mvdan/fdroidcl.(*Apk).URL(0xc4205d7c00, 0x0, 0x3)
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/index.go:203 +0x51
main.downloadApk(0xc4205d7c00, 0xc4200d6700, 0xc4205d7c00)
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/cmd/fdroidcl/download.go:40 +0x40
main.downloadAndDo(0xc420096018, 0x1, 0x1, 0xc4200d6700, 0x734f40)
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/cmd/fdroidcl/install.go:49 +0xc4
main.runInstall(0xc42007c1d0, 0x1, 0x1)
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/cmd/fdroidcl/install.go:35 +0x1cc
main.main()
	/tmp/yaourt-tmp-amo/aur-fdroidcl/src/src/github.com/mvdan/fdroidcl/cmd/fdroidcl/main.go:180 +0x384
```
Maybe I have to do something locally to use the newly downloaded version instead of the one installed from AUR? Or is there a way to find out what version of fdroidcl is being executed?

**Comment by mvdan** [comment]
@amo13 To check what version you're using, try `which fdroidcl`.

To force using the one that you installed via `go get`:

    $GOPATH/bin/fdroidcl install ...

The issue could be that you don't have `$GOPATH/bin` in your `$PATH`.

Also try removing the cache (again) once you're using the newest version.

**Comment by amo13** [comment]
Wow, hey man, you're amazing! Thank you a lot for your help with that, it works now.

**Comment by mvdan** [comment]
Happy to help :) will do a release soon.

**Comment by mvdan** [comment]
I've just released 0.3.1.
